### ISSUE TITLE
bracket nesting level exceeded maximum of 256 on macOS

### ISSUE THREAD

**Issue by kinosang** [ndless-nspire/Ndless#59]
While run `build_toolchain.sh`, it failed with `bracket nesting level exceeded maximum of 256`.

```
#define ATTRIBUTE_ASM_FPRINTF(m, n) __attribute__ ((__format__ (__asm_fp...
                                                    ^
../../gcc-6.2.0/gcc/config/arm/thumb1.md:1615:10873: fatal error: bracket
      nesting level exceeded maximum of 256
../../gcc-6.2.0/gcc/config/arm/thumb1.md:1615:10873: note: use -fbracket-depth=N
      to increase maximum nesting level
```

I tried to add ` -fbracket-depth=512` into `CFLAGS_FOR_TARGET` and `CXXFLAGS_FOR_TARGET` in `build_toolchain.sh`, but not works.

macOS Sierra 10.12.3
Homebrewed gcc-arm-none-eabi (px4/px4/gcc-arm-none-eabi: stable 20160928)
clang --version Apple LLVM version 8.0.0 (clang-800.0.42.1)

**Comment by Vogtinator** [comment]
What's your host toolchain version?

> I tried to add -fbracket-depth=512 into CFLAGS_FOR_TARGET and CXXFLAGS_FOR_TARGET in build_toolchain.sh, but not works.

That's the wrong place, you need to add it to ```CFLAGS```.

**Comment by kinosang** [comment]
and i didn't find where to add the `CFLAGS`

**Comment by adriweb** [comment]
This is actually a known issue, you can add it in the GCC options here: https://github.com/ndless-nspire/Ndless/blob/master/ndless-sdk/toolchain/build_toolchain.sh#L29
You should also remove the GDB's `--with-python` (at least last time I checked, which is quite a bit ago, but still, it was problematic)

**Comment by adriweb** [comment]
(I updated my comment above)

**Comment by kinosang** [comment]
@adriweb not work for me.
still show the error

```
...
../../gcc-6.2.0/gcc/config/arm/thumb1.md:1615:10873: fatal error: bracket
      nesting level exceeded maximum of 256
../../gcc-6.2.0/gcc/config/arm/thumb1.md:1615:10873: note: use -fbracket-depth=N
      to increase maximum nesting level
...
11 warnings generated.
9 warnings generated.
9 warnings generated.
9 warnings generated.
38 warnings generated.
47 warnings generated.
49 warnings generated.
rm gcc.pod
make[1]: *** [all-gcc] Error 2
make: *** [all] Error 2
```

**Comment by adriweb** [comment]
I'll try again when I'm back home tonight but considering it was working that way (the updated post, not the initial one), I don't see why it can't work anymore...

**Comment by kinosang** [comment]
@adriweb thank you very much. I found that there's two step of building gcc and the error is threw by step 2 now. I add the `CXXFLAGS` to it and it works well now.

The first step looks like `[...] && make $PARALLEL CXXFLAGS="-fbracket-depth=512" all-gcc && [...]` and the second step looks like `[...] && make $PARALLEL CXXFLAGS="-fbracket-depth=512" gcc && [...]`

**Comment by adriweb** [comment]
Fixed by #63
